creator-node/src/routes/tracks.js

```diff
@@ -886,31 +889,13 @@ router.get(
         debugTimings.findSourceFileForSegments =
           (Date.now() - findSourceFileForSegmentsStart) / 1000
 
-        // check that the number of files in the Files table for these segments for this user matches the number of segments from the metadata object
-        if (fileSegmentRecords.length !== trackRecord.track_segments.length) {
-          req.logger.warn(
-            `Track stream content mismatch for blockchainId ${blockchainId} - number of segments don't match between local and discovery`
-          )
-        }
-
-        // check that there's a single sourceFile that all File records share by getting an array of uniques
-        const uniqSourceFiles = fileSegmentRecords
-          .map((record) => record.sourceFile)
-          .filter((v, i, a) => a.indexOf(v) === i)
-
-        if (uniqSourceFiles.length !== 1) {
-          req.logger.warn(
-            `Track stream content mismatch for blockchainId ${blockchainId} - there's not one sourceFile that matches all segments`
-          )
-        }
-
         // search for the copy320 record based on the sourceFile
         const findCopy320AfterFallbackStart = Date.now()
         fileRecord = await models.File.findOne({
           attributes: ['multihash'],
           where: {
             type: 'copy320',
-            sourceFile: uniqSourceFiles[0]
+            sourceFile: fileSegmentRecords[0].sourceFile
```

Comment:
what's the significance of the 0th indexed element?

Reply:
\ will this basically only return 1 source file?

Reply:
we pick the first sourceFile as the one to play. nothing special about it, but basically we can't exactly deduce which sourceFile is the right one so we pick idx 0

Reply:
hm, in that case, can this just be a `limit 1` filter? looks like there might not be a need for the distinct source file filter?

Reply:
yes that is a great idea